api/authenticator.py

```diff
@@ -2685,37 +2685,61 @@ class BasicAuthTempTokenController(object):
     """A controller that handles requests for issuing temporary tokens
     to HTTP Basic Auth credentials.
     """
+    DO_NOT_GENERATE_NEW_TOKEN_PERIOD = 60 * 59
+    TOKEN_DURATION = datetime.timedelta(seconds=3600)
```

Comment:
Since the rule is "don't generate a token in the first minute of its lifetime", DO_NOT_GENERATE_NEW_TOKEN_PERIOD should be derived from TOKEN_DURATION, i.e. `TOKEN_DURATION.seconds - 60`. As it is, if TOKEN_DURATION changes, DO_NOT_GENERATE_NEW_TOKEN_PERIOD will stay the same and possibly end up making no sense.